doc/en/fixture.rst

```diff
@@ -400,6 +400,35 @@ The ``smtp_connection`` connection will be closed after the test finished
 execution because the ``smtp_connection`` object automatically closes when
 the ``with`` statement ends.
 
+Using the contextlib.ExitStack context manager finalizers will always be called
+regardless if the fixture *setup* code raises an exception. This is handy to properly
+close all resources created by a fixture even if one of them fails to be created/acquired:
+
+.. code-block:: python
+
+    # content of test_yield3.py
+
+    import contextlib
+
+    import pytest
+
+    from .utils import connect
+
+
+    @pytest.fixture
+    def equipments():
```

Comment:
I'm not sure this is worth adding at all, `request.addfinalizer` is actually more concise:

```python
    @pytest.fixture
    def equipments(request):        
        r = []
        for port in ("C1", "C3", "C28"):
            equip = connect(port)
            request.addfinalizer(equip.disconnect)
            r.append(equip)
        yield r
```

Given that we definitely won't deprecate `request.addfinalizer`, I don't see much value in adding this particular example TBH. 🤔

Reply:
It's more concise in this particular case but it doesn't let you use layered resources or mix contexts and callbacks.

ExitStack would be more concise considering realistically connect will return a context manager

Reply:
> ExitStack would be more concise considering realistically connect will return a context manager

Can you update the example to make use of this then? I believe it would then make more sense.

Reply:
done